### ISSUE TITLE
Cannot use Tab to move to columns

### ISSUE THREAD

**Issue by alexstine** [WordPress/gutenberg#41014]
Steps:
1. Insert a table block.
2. Select Create Table button.
3. Press Tab key to move to the second column.
4. Notice how focus leaves the block and arrives at the sidebar.
5. I expected focus to land in the second column and only reach the sidebar after the field labeled Caption.

This is related to writing flow. Just can't figure out how to adjust the logic here.

**Comment by costdev** [comment]
### Reproduction report
#### Environment
- OS: Windows
- Server: WSL2 (Ubuntu)
- Browser: Google Chrome 101.0.4951.54
- WordPress: 6.1-alpha-53344-src
- Gutenberg version: with and without Gutenberg 13.1.0 active

#### Steps to reproduce.
- Insert a table block.
- Select Create Table button.
- Press Tab key to move to the second column.
- 🐞Notice how focus leaves the block and arrives at the sidebar.

#### Results
- Issue reproduced. ✅

**Comment by t-hamano** [comment]
I developed [a table block plugin](https://wordpress.org/plugins/flexible-table-block/) with [tab key navigation](https://github.com/t-hamano/flexible-table-block/blob/main/src/elements/table.tsx#L229-L274).
This is not an extension of RichText, but rather the logic to detect keyboard events within a block.
I am sure there are many improvements that could be made, but I hope this will be helpful.

**Comment by ajaydsouza** [comment]
Can this be prioritised for implementation? It's intuitive across when working with editors to use Tab to navigate. Right now it takes you completely out of the table.

**Comment by t-hamano** [comment]
As far as I can tell, when a block or an editable element within a block is selected, pressing the Tab key focuses the next focusable element, such as a tab in the block sidebar.

I'm afraid that if we were to introduce our own tab focus management for the Table block, it would conflict with the current behavior.

**Comment by ajaydsouza** [comment]
I agree on Tab going to the next focusable element. In the context of a table, the next focus of the existing cell is the cell to the right of it. So it would respect the focus behaviour.

I came here yesterday to this thread as I was editing a table, in the middle of a post, hit tab and the cursor jumped to the end of the post I was editing.

This is what is @alexstine proposed above as well.
